src/providers/inventory-data.ts

```diff
@@ -17,28 +17,23 @@ export class InventoryData {
   }
 
   addItem(item: Object) {
-    return this.authHttp.put(ApiUrl + Links.item, item)
-      .map(this.extractData);
+    return this.putEndpoint(Links.item, item);
   }
 
   editItem(item: Object, tag: string) {
-    return this.authHttp.put(ApiUrl + Links.item + '/' + tag, item)
-      .map(this.extractData);
+    return this.putEndpoint(Links.item + '/' + tag, item);
```

Comment:
You could use template strings here (and in the functions below).
```JavaScript
return this.putEndpoint(`${Links.item}/${tag}`, item);
```
I tend to prefer template strings over constructing with `+`, but for no real reason. Do you prefer one syntax over the other?